llvm/lib/Target/AArch64/AArch64InstrInfo.td

```diff
@@ -5063,6 +5063,12 @@ def : Pat<(fmaxnum_ieee (f32 FPR32:$a), (f32 FPR32:$b)),
 def : Pat<(fmaxnum_ieee (f16 FPR16:$a), (f16 FPR16:$b)),
           (FMAXNMHrr FPR16:$a, FPR16:$b)>;
 
+def : Pat<(f16 (fcanonicalize f16:$a)),
+          (FMINNMHrr   f16:$a, f16:$a)>;
```

Comment:
FP16 patterns require a hasFullFP16 predicate, otherwise the instructions will not be available. Same for vector patterns.